core/src/test/java/GameObjects/UrlBuilderImpTest.java

```diff
@@ -0,0 +1,95 @@
+package GameObjects;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the UrlBuilder. The URLs provided were taken directly from the database using their
+ * API.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UrlBuilderImpTest {
```

Comment:
A lot of these tests test basically the same thing.  I.e., each of these test that the urlBuilder takes the return value of the settings object, and puts it in the correct place in the url.  Consider removing a few of the redundant tests, and adding some tests where the settings object returns an unexpected value, like "foo", an empty string, or null.